concordium-consensus/src/Concordium/Scheduler/Environment.hs

```diff
@@ -1047,11 +1050,16 @@ withDeposit wtc comp k = do
             -- (energy ticked so far).
             (usedEnergy, payment) <- computeExecutionCharge totalEnergyToUse (ls ^. energyLeft)
             chargeExecutionCost (wtc ^. wtcSenderAccount) payment
+            -- The exuction cost is payed for by the sponsor if present, otherwise by the sender.
+            let (tsCost, mbSponsorDetails)
+                    | Just sdSponsor <- (wtc ^. wtcSponsorAddress) =
+                        (0, Just SponsorDetails{sdCost = payment, ..})
+                    | otherwise = (payment, Nothing)
```

Comment:
I don't think this should be in scope for this PR. In particular, the logic isn't implemented to charge the sponsor account. Instead, just set the sponsor details to `Nothing` on the `TransactionSummary` for now.